nf_core/lint.py

```diff
@@ -657,6 +666,10 @@ def check_version_consistency(self):
         if self.config.get('process.container', ''):
             versions['process.container'] = self.config.get('process.container', '').strip(' \'"').split(':')[-1]
 
+        # Get version from the GITHUB_REF env var
+        if os.environ.get('GITHUB_REF') and os.environ.get('GITHUB_REPOSITORY', '') != 'nf-core/tools':
+            versions['GITHUB_REF'] = os.environ.get('GITHUB_REF').strip(' \'"')
```

Comment:
This might need a bit more testing. I think `GITHUB_REF` gives the full git reference, not just the branch name. For example `refs/tags/0.1.2`.  Hopefully stripping away everything up to the final `/` should work though.

Will be the same for the branch names.

Reply:
Ok. I have no idea what Im doing with these variables and how to test. Do you fancy looking at these bits in another PR? Hopefully, everything else should be ok now.

Reply:
Should be easy to remove if you fake it and pretend that they are directory separators I think. Just run `GITHUB_REF` through `os.path.basename`

Reply:
Thats what I was thinking of doing but I dont know exactly which bits of the path are required and where other bits are required in other places 😅

Reply:
See https://github.com/nf-core/tools/pull/502/commits/2906f68d17f0731d48d0fb4e23e6d11c88d1a383 ?

Reply:
Yup! Also needs an additional check though. Previous travis env variable was only there if it was a release, but the ref one is always there. So need to check if it `startswith` `/refs/tags/` too I think.

Reply:
Alternate method that might be nicer would be to use a custom environment variable to trigger this behaviour, then export that only in the correct circumstances using conditionals in the actions workflow. 

Make sense?

Reply:
Nope. Not really. I still dont completely understand how it all comes together which is why Im struggling to wrap my head around it...might have to have a fresh look next week 👍